lib/engine/game/g_18_usa/step/buy_sell_par_shares.rb

```diff
@@ -11,6 +11,44 @@ class BuySellParShares < G1817::Step::BuySellParShares
           MAX_BID = 100_000
           MAX_PAR_PRICE = 200
 
+          def corporate_actions(entity)
+            return [] if @winning_bid
+
+            return [] if @corporate_action && @corporate_action.entity != entity
+
+            actions = []
+            if @round.current_actions.none?
+              actions << 'take_loan' if @game.can_take_loan?(entity) && !@corporate_action.is_a?(Action::BuyShares)
+              actions << 'buy_shares' unless @game.redeemable_shares(entity).empty?
+              actions << 'scrap_train' if can_scrap_train?(entity)
+            end
+            actions
+          end
+
+          def can_scrap_train?(entity)
```

Comment:
Maybe use a module, so you don't have to duplicate these methods between here and the scrap_train step?

Reply:
Since `can_scrap_train?` is the same between `buy_sell_par_shares` and `scrap_train` I pulled it into the `scrap_train_module` but `process_scrap_train` is slightly different between the two (one calls `scrap_train_by_owner`, the other calls `scrap_train_by_corporation`)